lib/stm32/f4/rtc.c

```diff
@@ -29,11 +29,248 @@
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_TR & (RTC_TR_HT_MASK << RTC_TR_HT_SHIFT)) >> 20);
+  bcd.units = ((RTC_TR & (RTC_TR_HU_MASK << RTC_TR_HU_SHIFT)) >> 16);
+  val = rtc_bcd_to_dec(bcd);
+  return val;
+}
+
+static uint8_t rtc_get_minute(void)
+{
+  uint8_t val;
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_TR & (RTC_TR_MNT_MASK << RTC_TR_MNT_SHIFT)) >> 12);
+  bcd.units = ((RTC_TR & (RTC_TR_MNU_MASK << RTC_TR_MNU_SHIFT)) >> 8);
+  val = rtc_bcd_to_dec(bcd);
+  return val;
+}
+
+static uint8_t rtc_get_second(void)
+{
+  uint8_t val;
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_TR & (RTC_TR_ST_MASK << RTC_TR_ST_SHIFT)) >> 4);
+  bcd.units = ((RTC_TR & (RTC_TR_SU_MASK << RTC_TR_SU_SHIFT)) >> 0);
+  val = rtc_bcd_to_dec(bcd);
+  return val;
+}
+
+/* Date Register contains only 1 byte for storing Year.
+ * Year Register = (Year - 1970)
+ */
+static uint16_t rtc_get_year(void)
+{
+  uint16_t val;
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_DR & (RTC_DR_YT_MASK << RTC_DR_YT_SHIFT)) >> 20);
+  bcd.units = ((RTC_DR & (RTC_DR_YU_MASK << RTC_DR_YU_SHIFT)) >> 16);
+  val = rtc_bcd_to_dec(bcd) + 1970;
+  return val;
+}
+
+static uint8_t rtc_get_week_day(void)
+{
+  uint8_t val;
+  val = ((RTC_DR & (RTC_DR_WDU_MASK << RTC_DR_WDU_SHIFT)) >> 13);
+  return val;
+}
+
+static uint8_t rtc_get_month(void)
+{
+  uint8_t val;
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_DR & (RTC_DR_MT_MASK << RTC_DR_MT_SHIFT)) >> 12);
+  bcd.units = ((RTC_DR & (RTC_DR_MU_MASK << RTC_DR_MU_SHIFT)) >> 8);
+  val = rtc_bcd_to_dec(bcd);
+  return val;
+}
+
+static uint8_t rtc_get_day(void)
+{
+  uint8_t val;
+  struct bcd_s bcd;
+  bcd.tens = ((RTC_DR & (RTC_DR_DT_MASK << RTC_DR_DT_SHIFT)) >> 4);
+  bcd.units = ((RTC_DR & (RTC_DR_DU_MASK << RTC_DR_DU_SHIFT)) >> 0);
+  val = rtc_bcd_to_dec(bcd);
+  return val;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Configure RTC for using calendar based on the LSE clock
+    @warning You must configure RCC peripheral for RTC and PWR
+
+*/
+void rtc_calendar_config(void)
+{
+  pwr_disable_backup_domain_write_protect();
+  /* Enable LSE for using calendar */
+  RCC_BDCR |= RCC_BDCR_LSEON;
+  RCC_BDCR |= RCC_BDCR_RTCEN;
+  RCC_BDCR |= (1<<8); /* RTCSEL at 0b01 */
+  RCC_BDCR &= ~(1<<9); /* RTCSEL at 0b01 */
+
+  while(!(RCC_BDCR & RCC_BDCR_LSERDY));
+
+  pwr_enable_backup_domain_write_protect();
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Set calendar on the RTC register
+    @warning This function uses a time structure like in standard C
+*/
+void rtc_calendar_set(struct time_s time)
+{
+  pwr_disable_backup_domain_write_protect();
+  rtc_unlock();
+  /* Allow to update Calendar value */
+  RTC_ISR |= RTC_ISR_INIT;
+  while(!(RTC_ISR & RTC_ISR_INITF));
+
+  /* Set Time Register */
+  /* Be careful : TR register must be setting in one time */
+  uint32_t reg_tr = rtc_set_hour(time.hour)      + \
+                    rtc_set_minute(time.minute)  + \
```

Comment:
aside from comments on how this is being done, you don't need line continuations.